examples/cli/effects.roc

```diff
@@ -1,11 +1,27 @@
-app [main] { pf: platform "effects-platform/main.roc" }
+app [main!] { pf: platform "effects-platform/main.roc" }
 
-import pf.PlatformTasks
+import pf.Effect
 
-main : Task {} []
-main =
-    line = PlatformTasks.getLine!
-    PlatformTasks.putLine! "You entered: $(line)"
-    PlatformTasks.putLine! "It is known"
+main! : {} => {}
+main! = \{} ->
+    ["Welcome!", "What's your name?"]
+    |> forEach! Effect.putLine!
 
-    Task.ok {}
+    line = Effect.getLine! {}
+
+    if line == "secret" then
+        Effect.putLine! "You found the secret"
+        Effect.putLine! "Congratulations!"
+    else
+        {}
+
+    Effect.putLine! "You entered: $(line)"
+    Effect.putLine! "It is known"
+
+forEach! : List a, (a => {}) => {}
```

Comment:
Do you plan on adding `List.walk!` or even `List.forEach!` to the std lib in this PR? If not, we'll want to make an issue to do so. I believe we decided in Zulip we'd like one or both of those in the meantime while `for` loops are cooking.

Reply:
Yeah, I think we should introduce that separately. I was going to do so, but I think we need to discuss the API design.  The one I wrote in this example wouldn't be very useful in real platforms because most effectful functions would return a `Result`. I'll start a thread on Zulip.